### PR TITLE
Home effects

### PR DESCRIPTION
**Pull request by davepl**
Among other changes, this checkin adds the features around my house for testing purposes.  That way you get a half dozen canvases and a dozen features running in parallel for testing.  

This will be replaced by a JSON config file load soon enough, so for now please comment out whatever in LoadCanvas() you do not want.

### CODE REVIEW

File: effectsmanager.h [hunk 1]
```diff
@@ -157,6 +150,12 @@ class EffectsManager : public IEffectsManager
                     }
                 }
 
+                auto now = steady_clock::now();
+                while (now < nextFrameTime && _running) {
```
Comment (davepl) [inline effectsmanager.h:154]: I should have commented that the intent is to wait while checking _running every 10ms or so

File: main.cpp [hunk 2]
```diff
@@ -101,16 +97,209 @@ vector<unique_ptr<ICanvas>> LoadCanvases()
...
+            49152,                // Port∏
+            length4, 1,           // Width, Height
+            start4, 0,            // Offset X, Offset Y
+            false,                // Reversed
+            0,                    // Channel
+            false,                // Red-Green Swap
+            21
+        );
+        
+        //canvasCabana->AddFeature(std::move(featureCabana1));
```
Comment (davepl) [inline main.cpp:295]: Commented out because it's down right now, but that's the only reason

File: sample_config.json [hunk 3]
```diff
@@ -0,0 +1,222 @@
+[
```
Comment (davepl) [inline sample_config.json:1]: Not quite complete yet, but this will be approximately the persisted settings form

**Review by rbergen — COMMENTED** [review]
I've noticed that with my Mesmerizer, the current ndscpp (i.e. without this PR applied) seems to lose the connection after some time, and then doesn't recover. In fact, if I then try to shutdown ndscpp by pressing Ctrl-C it will lock up after the webserver exits, and I have to kill it with a SIGKILL from another terminal window.

All of this leads to logging like this: 

```log
Sending 5 packets, 26234 bytes at queue size 0
Sending 5 packets, 26977 bytes at queue size 0
Sending 5 packets, 26305 bytes at queue size 0
Sending 10 packets, 53139 bytes at queue size 0
Sending 11 packets, 59448 bytes at queue size 0
Sending 14 packets, 74271 bytes at queue size 0
Sending 13 packets, 68332 bytes at queue size 0
Sending 15 packets, 78943 bytes at queue size 0
Sending 19 packets, 97736 bytes at queue size 0
Sending 20 packets, 100392 bytes at queue size 6
Sending 20 packets, 108100 bytes at queue size 42
Queue is full, dropping frame and resetting socket
Closing socket connection to 192.168.1.138
^C(2024-11-27 07:24:56) [INFO    ] Closing IO service 0x55beb7384930
(2024-11-27 07:24:56) [INFO    ] Closing IO service 0x55beb7384938
(2024-11-27 07:24:56) [INFO    ] Closing IO service 0x55beb7384940
(2024-11-27 07:24:56) [INFO    ] Closing IO service 0x55beb7384948
(2024-11-27 07:24:56) [INFO    ] Closing IO service 0x55beb7384950
(2024-11-27 07:24:56) [INFO    ] Closing IO service 0x55beb7384958
(2024-11-27 07:24:56) [INFO    ] Closing IO service 0x55beb7384960
(2024-11-27 07:24:56) [INFO    ] Closing main IO service (0x55beb73f8ac8)
(2024-11-27 07:24:56) [INFO    ] Exiting.
Shutting down...
Killed
```

Should the changes in socketchannel.h that are included in this PR address this? If so, that's something I can test then.

Beyond that, LGTM.

**Pull request merged**